daemon/container_operations_unix.go

```diff
@@ -120,19 +122,21 @@ func (daemon *Daemon) setupIpcDirs(c *container.Container) error {
 				return err
 			}
 
-			shmSize := int64(daemon.configStore.ShmSize)
-			if c.HostConfig.ShmSize != 0 {
-				shmSize = c.HostConfig.ShmSize
-			}
-			shmproperty := "mode=1777,size=" + strconv.FormatInt(shmSize, 10)
+			shmproperty := "mode=1777,size=" + strconv.FormatInt(c.HostConfig.ShmSize, 10)
 			if err := unix.Mount("shm", shmPath, "tmpfs", uintptr(unix.MS_NOEXEC|unix.MS_NOSUID|unix.MS_NODEV), label.FormatMountLabel(shmproperty, c.GetMountLabel())); err != nil {
 				return fmt.Errorf("mounting shm tmpfs: %s", err)
 			}
 			if err := os.Chown(shmPath, rootIDs.UID, rootIDs.GID); err != nil {
 				return err
 			}
+			c.ShmPath = shmPath
 		}
 
+	case ipcMode.IsPrivate():
+		c.ShmPath = ""
+
+	default:
```

Comment:
Got it, thank you so much for pointing this out, I totally overlooked this scenario :-\

By the way, is it possible to change existing container's IpcMode (other than the direct editing of its hostconfig.json)? I am thinking about something like `docker set` which can be used to apply parameters to an existing container (surely, some can be applied runtime, some others like --ipc should require restart).